### PR TITLE
DEP: Deprecate the pickle aliases

### PR DESCRIPTION
**Pull request by eric-wieser**
* The np.ma functions are misleading, as they do not actually do anything special for ma.array
* The np.loads functions doesn't even have numpy-specific documentation, and does not behave consistently with `np.load`
* The string overloads of np.ma.load and np.ma.dump do not work well on python 3, as they make assumptions about whether a binary or text pickle file is used (gh-5491)

### CODE REVIEW

File: numpy/ma/core.py [hunk 1]
```diff
@@ -7859,6 +7859,16 @@ def asanyarray(a, dtype=None):
 ##############################################################################
 #                               Pickling                                     #
 ##############################################################################
+
+def _pickle_warn(method):
+    # NumPy 1.15.0, 2017-12-10
+    warnigns.warn(
```
Comment (charris) [inline numpy/ma/core.py]: Typo, warnigns.

**Comment by eric-wieser** [comment]
Fixed typo, and also deprecated `np.core.numeric.load` (which is different from `np.load`)

**Review by mhvk — APPROVED** [review]
Yes, nice to get rid of these.

File: doc/release/1.15.0-notes.rst [hunk 2]
```diff
@@ -14,6 +14,15 @@ New functions
 Deprecations
 ============
 
+* Aliases of builtin `pickle` functions are deprecated, in favor of their
+  unaliased ``pickle.<func>`` names:
+   * `np.loads`
```
Comment (charris) [inline doc/release/1.15.0-notes.rst]: Sublist needs to be preceded by blank line.

**Comment by charris** [comment]
I'll fix up the missing blank line and merge.

**Comment by charris** [comment]
Thanks Eric.

**Pull request merged**